### PR TITLE
Increace test coverage

### LINKED ISSUE #2993: 服务端渲染测试方案
有问题的组件：

- [x] **Mention**  https://github.com/facebook/draft-js/issues/385
- [x] **Menu**  https://github.com/react-component/menu/issues/63
- [x] **Upload** https://github.com/react-component/upload/issues/55

### PR DESCRIPTION
**Pull request by afc163**
https://github.com/facebook/draft-js/issues/702
https://github.com/facebook/draft-js/issues/385

close https://github.com/ant-design/ant-design/issues/2993

### CODE REVIEW

**Comment by yesmeck** [comment]
https://github.com/facebook/draft-js/issues/702#issuecomment-283169324 这个看起来可以吧

**Comment by afc163** [comment]
对，用的就是这个，现在的问题是 dist 还是会挂。貌似 jest mock 对 dist 不生效？

**Comment by yesmeck** [comment]
哦，对的 dist 没法 mock。

File: components/mention/__tests__/demo.test.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
 import demoTest from '../../../tests/shared/demoTest';
 
-demoTest('mention', { skip: true });
+jest.mock('draft-js/lib/generateRandomKey', () => () => '123');
```
Comment (afc163) [inline components/mention/__tests__/demo.test.js]: https://travis-ci.org/ant-design/ant-design/jobs/381997418

**Comment by yesmeck** [comment]
dist 就跳过这个测试吧

**Comment by afc163** [comment]
见鬼了 https://travis-ci.org/ant-design/ant-design/jobs/382132485

**Comment by afc163** [comment]
```
Summary of all failing tests
 FAIL  components/mention/__tests__/index.test.js
  ● Mention › change suggestions
    expect(received).toBe(expected) // Object.is equality
    
    Expected value to be:
      2
    Received:
      0
      50 |     const ed = wrapper.find('.public-DraftEditor-content');
      51 |     ed.simulate('beforeInput', { data: '@' });
    > 52 |     expect(wrapper.find('Nav').length).toBe(2);
      53 |     expect(wrapper.find('Nav').first().text()).toBe('afc163');
      54 |     wrapper.setState({ suggestions: ['yesmeck', 'yiminghe', 'lucy'] });
      55 |     expect(wrapper.find('Nav').length).toBe(3);
      
      at Object.<anonymous> (components/mention/__tests__/index.test.js:52:40)
```

File: components/mention/__tests__/index.test.js [hunk 2]
```diff
@@ -43,18 +43,19 @@ describe('Mention', () => {
...
+      <Mention suggestions={['afc163', 'raohai']} getSuggestionContainer={() => container.getDOMNode()} />
     );
     wrapper.find('DraftEditorContents').simulate('focus');
     const ed = wrapper.find('.public-DraftEditor-content');
     ed.simulate('beforeInput', { data: '@' });
     jest.runAllTimers();
-    expect(wrapper.find('Nav').length).toBe(2);
-    expect(wrapper.find('Nav').first().text()).toBe('afc163');
+    expect(container.getDOMNode().querySelectorAll('.ant-mention-dropdown-item').length).toBe(2);
+    expect(container.getDOMNode().querySelectorAll('.ant-mention-dropdown-item')[0].innerHTML).toBe('afc163');
```
[the remark this answers is not in the document]
Comment (afc163) [reply components/mention/__tests__/index.test.js]: 是 React 15 下会通不过。

**Pull request merged**